duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
-        //sr.CVNResultNutau = v[i::kNutauQE] + v[i::kNutauRes] + v[i::kNutauDIS] + v[i::kNutauOther]
-
-        sr.CVNResultIsAntineutrino = (*cvnin)[0].GetIsAntineutrinoProbability();
-
-        sr.cvnnue = (*cvnin)[0].GetNueProbability();
-        sr.cvnnumu = (*cvnin)[0].GetNumuProbability();
-        sr.cvnnutau = (*cvnin)[0].GetNutauProbability();
-        sr.cvnnc = (*cvnin)[0].GetNCProbability();
-
-        sr.CVNResult0Protons = (*cvnin)[0].Get0protonsProbability();
-        sr.CVNResult1Protons = (*cvnin)[0].Get1protonsProbability();
-        sr.CVNResult2Protons = (*cvnin)[0].Get2protonsProbability();
-        sr.CVNResultNProtons = (*cvnin)[0].GetNprotonsProbability();
-
-        sr.CVNResult0Pions = (*cvnin)[0].Get0pionsProbability();
-        sr.CVNResult1Pions = (*cvnin)[0].Get1pionsProbability();
-        sr.CVNResult2Pions = (*cvnin)[0].Get2pionsProbability();
-        sr.CVNResultNPions = (*cvnin)[0].GetNpionsProbability();
-
-        sr.CVNResult0Pizeros = (*cvnin)[0].Get0pizerosProbability();
-        sr.CVNResult1Pizeros = (*cvnin)[0].Get1pizerosProbability();
-        sr.CVNResult2Pizeros = (*cvnin)[0].Get2pizerosProbability();
-        sr.CVNResultNPizeros = (*cvnin)[0].GetNpizerosProbability();
-
-        sr.CVNResult0Neutrons = (*cvnin)[0].Get0neutronsProbability();
-        sr.CVNResult1Neutrons = (*cvnin)[0].Get1neutronsProbability();
-        sr.CVNResult2Neutrons = (*cvnin)[0].Get2neutronsProbability();
-        sr.CVNResultNNeutrons = (*cvnin)[0].GetNneutronsProbability();
+  //------------------------------------------------------------------------------
+
+  void CAFMaker::FillEnergyInfo(caf::SRNeutrinoEnergyBranch &ErecBranch, const art::Event &evt) const
+  {
+    //Filling the reg CNN results
+    art::InputTag itag(fRegCNNLabel, "regcnnresult");
+    art::Handle<std::vector<cnn::RegCNNResult>> regcnn = evt.getHandle<std::vector<cnn::RegCNNResult>>(itag);
+    if(!regcnn.failedToGet() && !regcnn->empty()){
+        const std::vector<float>& cnnResults = (*regcnn)[0].fOutput;
+        ErecBranch.regcnn = cnnResults[0];
+    }
+    else{
+      std::cout << "Warning: " << itag << " does not correspond to a valid RegCNNResult product" << std::endl;
+    }
+
+    art::Handle<dune::EnergyRecoOutput> ereconue = evt.getHandle<dune::EnergyRecoOutput>(fEnergyRecoNueLabel);
+    art::Handle<dune::EnergyRecoOutput> ereconumu = evt.getHandle<dune::EnergyRecoOutput>(fEnergyRecoNumuLabel);
+
+
+    if(ereconue.failedToGet()){
+      std::cout << "Warning: " << fEnergyRecoNueLabel << " does not correspond to a valid EnergyRecoOutput product" << std::endl;
+    }
+    else{
+      ErecBranch.calo = ereconue->fNuLorentzVector.E();
+    }
+
+    if(ereconumu.failedToGet()){
+      std::cout << "Warning: " << fEnergyRecoNumuLabel << " does not correspond to a valid EnergyRecoOutput product" << std::endl;
+    }
+    else{
+      ErecBranch.lep_calo = ereconumu->fNuLorentzVector.E();
+    }
+
```

Comment:
I was confused why the numu hyp was being used to get the lepton energy, and the nue hyp the neutrino energy. I can see it's most likely because the standard record object doesn't have more general member variables? i.e. it only saves one reco nu energy. 

As you know, the old one stores both the nue and numu hyp energies. This then allows you to investigate different selections at the CAF stage. Can these params be added to the standard record? This is likely a question for jeremy?

Reply:
I think the information filled in the end are the same, this is mostly the naming convention that changed. `calo` energy corresponds to the energy estimation for nu_e while `lep_calo` energy corresponds to the energy estimation for nu_mu.